From 1add2d16cd4dcb814ec917dbad0a0435ed0b6e73 Mon Sep 17 00:00:00 2001
From: gregor herrmann <gregoa@debian.org>
Date: Fri, 9 Dec 2011 00:55:55 +0100
Subject: [PATCH] dh_perl: add -S (_s_ameperl) switch in v9

The -S switch will add e.g. "perl (>= 5.14.2)" and "perl (<< 5.14.3~)" to
the ${perl:Depends} variable for packages that need the exact perl version
they are built against.

Closes: #579112
---
 dh_perl |   16 +++++++++++++++-
 1 files changed, 15 insertions(+), 1 deletions(-)

diff --git a/dh_perl b/dh_perl
index bf2506c..9977ad4 100755
--- a/dh_perl
+++ b/dh_perl
@@ -56,6 +56,12 @@ directories, you can make B<dh_perl> check those directories by passing their
 names on the command line. It will only check the F<vendorlib> and F<vendorarch>
 directories by default.
 
+=item B<-S>
+
+Starting with compatibility level 9, B<${perl:Depends}> adds B<perl[-base]
+(E<gt>= $currentversion)> and B<perl[-base] (<< $nextminorversion~)> to
+Depends, if B<dh_perl> is invoked with B<-S> (mnemonic: _s_ame perl version).
+
 =back
 
 =head1 CONFORMS TO
@@ -66,7 +72,7 @@ Perl policy, version 1.20
 
 =cut
 
-init();
+init(options => { S => \$dh{S_FLAG} });
 
 my $vendorlib = substr $Config{vendorlib}, 1;
 my $vendorarch = substr $Config{vendorarch}, 1;
@@ -131,6 +137,14 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 		addsubstvar($package, "perl:Depends",
 			"perlapi-" . ($Config{debian_abi} || $Config{version}))
 			if $deps & XS_MODULE;
+
+		if (! compat(8) && $dh{S_FLAG}) {
+			my $currentversion = ">= $Config{version}";
+			my ($v1, $v2, $v3) = split /\./, $Config{version};
+			my $nextversion = "<< " . (join ".", $v1, $v2, $v3 + 1) . "~";
+			addsubstvar($package, "perl:Depends", $perl, $currentversion);
+			addsubstvar($package, "perl:Depends", $perl, $nextversion);
+		}
 	}
 
 	# MakeMaker always makes lib and share dirs, but typically
-- 
1.7.7.3

